go-controller/pkg/ovn/controller/apbroute/external_controller.go

```diff
@@ -0,0 +1,239 @@
+package apbroute
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	adminpolicybasedrouteapi "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/adminpolicybasedroute/v1"
+	adminpolicybasedroutelisters "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/adminpolicybasedroute/v1/apis/listers/adminpolicybasedroute/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	ktypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+)
+
+type gatewayInfoList []*gatewayInfo
+
+func (g gatewayInfoList) String() string {
+
+	s := strings.Builder{}
+	for _, item := range g {
+		s.WriteString(fmt.Sprintf("%s, ", item.gws))
+	}
+	return s.String()
+}
+
+func (g gatewayInfoList) HasIP(ip string) bool {
+	for _, i := range g {
+		if i.gws.Has(ip) {
+			return true
+		}
+	}
+	return false
+}
+
+func (g gatewayInfoList) Insert(items ...*gatewayInfo) gatewayInfoList {
+	for _, item := range items {
+		for _, ip := range item.gws.UnsortedList() {
+			if g.HasIP(ip) {
+				klog.Warningf("Duplicated static IP %s", ip)
+				continue
+			}
+			g = append(g, item)
+		}
+	}
+	return g
+}
+func (g gatewayInfoList) Delete(item *gatewayInfo) gatewayInfoList {
+	for index, i := range g {
+		if i.gws.Equal(item.gws) {
+			g = append(g[:index], g[index+1:]...)
+		}
+	}
+	return g
+}
+
+func (g gatewayInfoList) Len() int {
+	return len(g)
+}
+
+func (g gatewayInfoList) Less(i, j int) bool { return lessGWSIP(g[i], g[j]) }
```

Comment:
where is this function used?

Reply:
No, you still need to run the `sort.Sort` call if you want the object elements to be sorted. Turns out there are [3 ways for sorting](https://yourbasic.org/golang/how-to-sort-in-go/).
1) Use predefined sort functions for primitives (not our case)
2) Sort with custom comparators
3)  Sort implementing the `sort.Interface` interface. 

I went for the last one but the second one could have made the trick from what I can see.

Let me know what you think...